editor/src/components/canvas/controls/selection-area-helpers.ts

```diff
@@ -0,0 +1,132 @@
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { mapDropNulls } from '../../../core/shared/array-utils'
+import * as EP from '../../../core/shared/element-path'
+import { ElementInstanceMetadataMap } from '../../../core/shared/element-template'
+import {
+  CanvasRectangle,
+  WindowPoint,
+  canvasRectangle,
+  isFiniteRectangle,
+  rectangleContainsRectangle,
+} from '../../../core/shared/math-utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+
+type ElementUnderSelectionAreaType = 'scene' | 'regular'
+
+type ElementUnderSelectionArea = {
+  path: ElementPath
+  type: ElementUnderSelectionAreaType
+  fullyContained: boolean
+  selected: boolean
+}
+
+function getElementUnderSelectionAreaType(
+  metadata: ElementInstanceMetadataMap,
+  path: ElementPath,
+): ElementUnderSelectionAreaType | null {
+  if (MetadataUtils.isProbablyScene(metadata, path)) {
+    return 'scene'
+  }
+  return 'regular'
+}
+
+function elementIsFullyContainedInArea(
+  area: CanvasRectangle,
+  path: ElementPath,
+  metadata: ElementInstanceMetadataMap,
+): boolean {
+  const element = MetadataUtils.findElementByElementPath(metadata, path)
+  return element == null || element.globalFrame == null || !isFiniteRectangle(element.globalFrame)
+    ? false
+    : rectangleContainsRectangle(area, element.globalFrame)
+}
+
+export const filterUnderSelectionArea = (
+  paths: ElementPath[],
+  metadata: ElementInstanceMetadataMap,
+  area: CanvasRectangle | null,
+  selectedViews: ElementPath[],
+): ElementPath[] => {
+  if (area == null) {
+    return []
+  }
+  const elements: ElementUnderSelectionArea[] = mapDropNulls((path) => {
+    const type = getElementUnderSelectionAreaType(metadata, path)
+    if (type == null) {
+      return null
+    }
+    return {
+      path: path,
+      type: type,
+      fullyContained: elementIsFullyContainedInArea(area, path, metadata),
+      selected: EP.containsPath(path, selectedViews),
+    }
+  }, paths)
+
+  const isSceneChild =
+    (element: ElementUnderSelectionArea) => (other: ElementUnderSelectionArea) => {
+      return other.type === 'scene' && EP.isDescendantOf(element.path, other.path)
+    }
+
+  const thereAreStoryboardChildren = elements.some(
+    (element) => element.type === 'regular' && !elements.some(isSceneChild(element)),
+  )
+
+  return elements
+    .filter((element) => {
```

Comment:
There is something still not quite right about this algorithm. In the default project I can single click to select the `div` containing the Utopia logo when nothing is selected, but when using the area select I can only select the `Playground` component (which shouldn't be selectable) 
<img width="126" alt="image" src="https://github.com/concrete-utopia/utopia/assets/1044774/4298a2a3-480f-4daa-a888-606850119c5d">

Reply:
this https://github.com/concrete-utopia/utopia/pull/3750/commits/b5d9eebf4a6499412c1fe996c0a23fbe84ea42dc should solve this, I included the same selectable views used in the regular select mode in the calculations (before the selection area-specific filters)